### PR TITLE
fix(circular-dependency): Remove warning about circular dependency workspaceService-->devFileService --> workspaceService

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
remove circular dependency

### What issues does this PR fix or reference?
Circular dependency

### How to test this PR?
Should work in dev Workspace mode.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next

### CODE REVIEW

**Comment by benoitf** [comment]
[crw-ci-test]

**Comment by benoitf** [comment]
[crw-ci-test]

File: extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-devworkspace-env-variables.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      this.workspaceNamespace = process.env.DEVWORKSPACE_NAMESPACE;
+    }
+    if (process.env.DEVWORKSPACE_NAME === undefined) {
+      console.error('Environment variable DEVWORKSPACE_NAME is not set');
+    } else {
+      this.workspaceName = process.env.DEVWORKSPACE_NAME;
+    }
+  }
+
+  getWorkspaceId() {
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-devworkspace-env-variables.ts:51]: Is there a reason to omit the return types here?

**Review by azatsarynnyy — COMMENTED** [review]
I'm approving the PR but can't check it on my DevWorkspace installation as it seems broken now:
`
Error from server (InternalError): error when creating "flattened_theia-next.yaml": Internal error occurred: failed calling webhook "mutate.devworkspace-controller.svc": Post "https://devworkspace-webhookserver.devworkspace-controller.svc:443/mutate?timeout=30s": dial tcp 10.107.143.201:443: connect: connection refused
`

Obviously, it's not related to this PR :slightly_smiling_face:

**Review by azatsarynnyy — APPROVED** [review]

File: extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-devworkspace-env-variables.ts (hunk 1 above)
Comment (benoitf) [reply extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-devworkspace-env-variables.ts]: no, added
I think we had a lint error when we were missing them

**Comment by benoitf** [comment]
@azatsarynnyy could you try to run first `make install_cert_manager` in your minikube/kubernetes instance ?

**Comment by azatsarynnyy** [comment]
> @azatsarynnyy could you try to run first `make install_cert_manager` in your minikube/kubernetes instance ?

@benoitf I'm not sure I understand how to apply that command. Where should it be executed?

**Comment by azatsarynnyy** [comment]
Considering it works well on the DevWorkspace engine for @benoitf, the tests are green for the Che Server engine
I think it's good to merge.

**Comment by benoitf** [comment]
merging, happy path was green on
```
Happy Path / happy-path (alpine, stable) (pull_request) 
```

**Comment by azatsarynnyy** [comment]
I figured that out. It wasn't a problem with my Che installation, but with my `flattened_theia-next.yaml`. It was not valid:
> ValidationError(DevWorkspace.spec.template.components[2].volume): unknown field "ephemeral" in io.devfile.workspace.v1alpha2.DevWorkspace.spec.template.components.volume

**Pull request merged**